context/index.js

```diff
@@ -11,50 +12,69 @@ export const AppContextProvider = (props) => {
   const [triviaData, setTriviaData] = useState([]);
-  const [id, setId] = useState();
+  const [categoryId, setCategoryId] = useState();
   const [difficulty, setDifficulty] = useState();
   const [token, setToken] = useState();
 
   const getAllCategoryData = async () => {
-    Axios.get('https://opentdb.com/api_category.php').then(async (response) => {
-      const transformedCatagories = [];
-      response.data.trivia_categories.forEach((cat) => {
-        cat.name = cat.name.replace('Entertainment: ', '');
-        cat.name = cat.name.replace('Science: ', '');
-        transformedCatagories.push(cat);
-      });
-      setCategoryData(transformedCatagories);
-    });
+    await Axios.get('https://opentdb.com/api_category.php').then(
+      async (response) => {
+        const transformedCatagories = [];
+        response.data.trivia_categories.forEach((cat) => {
+          cat.name = cat.name.replace('Entertainment: ', '');
+          cat.name = cat.name.replace('Science: ', '');
+          transformedCatagories.push(cat);
+        });
+        setCategoryData(transformedCatagories);
+      },
+    );
   };
 
   //OpenDB Token so we do not get the same question twice, sessions are deleted after 6 hours
-  useEffect(() => {
-    const getToken = async () => {
-      Axios.get('https://opentdb.com/api_token.php?command=request').then(
-        (res) => {
+  const getToken = async () => {
+    if (token) {
+      await Axios.get(
+        `https://opentdb.com/api_token.php?command=reset&token=${token}`,
+      ).then((response) => {
+        setToken(response.data.token);
+      });
+    } else {
+      await Axios.get('https://opentdb.com/api_token.php?command=request').then(
+        async (res) => {
           if (res.data.response_code === 0) {
             setToken(res.data.token);
-          } else {
-            Axios.get(
-              `https://opentdb.com/api_token.php?command=reset&token=${token}`,
-            ).then((response) => {
-              setToken(response.data.token);
-            });
           }
         },
       );
-    };
+    }
+  };
+
+  useEffect(() => {
     getToken();
   }, []);
 
-  const getTriviaData = async (id, difficulty) => {
-    Axios.get(
-      `https://opentdb.com/api.php?amount=1&encode=base64&category=${id}&difficulty=${difficulty}&token=${token}`,
-    ).then((res) => {
-      setId(id);
-      setDifficulty(difficulty);
-      setTriviaData(res.data.results);
-    });
+  const getTriviaData = async (catId, diff) => {
+    setCategoryId(catId);
+    setDifficulty(diff);
+    await Axios.get(
+      `https://opentdb.com/api.php?amount=10&encode=base64&category=${catId}&difficulty=${diff}&token=${token}`,
+    )
+      .then(async (res) => {
+        if (res.data.response_code === 1) {
+          // not enough questions, need to think how to handle?
+          // seems to happen on celeb category on hard
+        }
+        if (res.data.response_code === 4) {
```

Comment:
Found that if I would re-use the same category and difficulty over and over I eventually ran out of question and would need to refresh the token. Hence, I moved the getToken out of the useEffect so I could re-use it myself. This error handling is still not perfect.

Reply:
Nice catch, yeah unfortunately some categories are pretty slim. Would be cool to build our own API buuut that's a lot of work lol